### PR TITLE
Port configuration from solium config keep

### PR DESCRIPTION
**Pull request by tomaszslabon**
This pull request contains:
- initial configuration for the `solhint-config-keep` (e.g. `README`, `.gitignore`)
- ports configuration rules from [solium-config-keep](https://github.com/keep-network/solium-config-keep)

Note: As [solium(ethlint)](https://github.com/duaraghav8/Ethlint) is no longer maintained and couldn't handle projects using newer versions of Solidity, we decided to replace it with [solhint](https://github.com/protofire/solhint).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// The rules below uses the recommended rules from solhint with some
+// modifications
+module.exports = {
+  "extends": "solhint:recommended",
+  'rules': {
+    "imports-on-top": "error",                     // recommended uses warn
+    "use-forbidden-name": "error",                 // recommended uses warn
+    "no-unused-vars": "error",                     // recommended uses warn
+    "reason-string": ["error", {"maxLength":250}], // recommended uses warn
+    "max-line-length": ["error", 145],             // not present in recommended
```
Comment (pdyraga) [inline index.js]: Why 145 and not 80?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "constructor-syntax": "error",                 // recommended uses warn
+    "visibility-modifier-order": "error",          // recommended uses warn
+    "ordering": "error",                           // not used in recommended
+    "var-name-mixedcase": "off",
+    "contract-name-camelcase": "off",
+    "const-name-snakecase": "off",
+    "compiler-version": "off"
```
Comment (pdyraga) [inline index.js]: I do not see this one in https://github.com/keep-network/solium-config-keep/blob/master/index.js. Why do we add it?

File: index.js (hunk 1 above)
Comment (tomaszslabon) [reply index.js]: The `solium` version used `max-len` with a default value, which was 145, but I can change it to 80.

File: index.js (hunk 2 above)
Comment (tomaszslabon) [reply index.js]: The `solium` version didn't check for the compiler version at all, that's why I turned it off (I wanted to get as close as possible to the original version).

We can enable this compiler version check, but we need to choose some value ([default version used by solhint is 0.5.8](https://github.com/protofire/solhint/blob/master/docs/rules/security/compiler-version.md)).  

I suggest we used some low version requirement (e.g.">=0.5.0") and individual projects may override this setting to a higher version.

File: index.js (hunk 1 above)
Comment (pdyraga) [reply index.js]: After thinking more about it, I'd drop this line completely and left the line length to `prettier` that will be invoked right after `solhint` in `lint:sol` and `lint:fix:sol` commands.

File: index.js (hunk 2 above)
Comment (pdyraga) [reply index.js]: Cool, your reasoning makes sense, let's leave it as-is. Especially that we have projects ranging from 0.5 to 0.8.

File: index.js (hunk 1 above)
Comment (tomaszslabon) [reply index.js]: Ok, I removed the `max-line-length` check completely.

File: index.js (hunk 2 above)
Comment (tomaszslabon) [reply index.js]: Ok, I left the `compiler-version` check `off`ed

File: index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+"use strict"
+
+// The list of rules that can be used in Solhint can be found at:
+// https://protofire.github.io/solhint/docs/rules.html
+
+// The rules below use the recommended rules from solhint with some
+// modifications
+module.exports = {
+  "extends": "solhint:recommended",
```
Comment (tomaszslabon) [inline index.js]: @pdyraga 
I think I found an issue here. 
I used this sharable configuration in one of our project and only the rules listed explicitly below are applied there
(all the rules from `recommended` ruleset were lost, e.g. the rules `quotes` and `no-empty-blocks` which are enabled by default in recommended). 

It looks like extending cannot be used in a sharable configuration like ours.

There are two things we can do:
1) List all the rules we want to use explicitly in this sharable (without extending `recommended`) - the best option in my opinion
2) Extend from both `recommended` and  this sharable config in our projects, like that:
```
{
  "extends": ["solhint:recommended", "keep"]
}
```

I think the first solution is the best one.

Comment (pdyraga) [reply index.js]: Yes, agreed, let's go with the first option.

**Review by pdyraga — APPROVED** [review]

**Pull request merged**